schema/encoding/jsonschema.go

```diff
@@ -0,0 +1,110 @@
+package jsonschema
+
+import (
+	"fmt"
+	"github.com/rs/rest-layer/schema"
+	"strconv"
+	"strings"
+)
+
+// JSONSchemaPropertyBase common Embedded Values
+type JSONSchemaPropertyBase struct {
+	Title string `json:"title"`
+	Type  string `json:"type"`
+}
+
+// JSONSchema represents a standard JSONSchema
+type JSONSchema struct {
+	JSONSchemaPropertyBase
+	Properties map[string]map[string]interface{} `json:"properties"`
+}
+
+// BoundariesToJSONSchema render boundaries to JSON
+func BoundariesToJSONSchema(b *schema.Boundaries) string {
+	if b == nil {
+		return ""
+	}
+	return fmt.Sprintf("%q: %s, %q: %s",
+		"minimum", strconv.FormatFloat(b.Min, 'E', -1, 64),
+		"maximum", strconv.FormatFloat(b.Max, 'E', -1, 64))
+}
+
+// ValidatorToJSONSchema takes a validator and renders to JSON
+func ValidatorToJSONSchema(v schema.FieldValidator) string {
+	if v == nil {
+		return ""
+	}
+	var result string
+	switch t := v.(type) {
+	case *schema.String:
+		result += fmt.Sprintf("%q: %q, ", "type", "string")
+		if t.Regexp != "" {
+			result += fmt.Sprintf("%q: %q, ", "pattern", t.Regexp)
+		}
+		if len(t.Allowed) > 0 {
+			var allowed []string
+			for _, value := range t.Allowed {
+				allowed = append(allowed, fmt.Sprintf("%q", value))
+			}
+			result += fmt.Sprintf("%q: [ %s ],", "oneOf", strings.Join(allowed, ","))
+		}
+		if t.MinLen > 0 {
+			result += fmt.Sprintf("%q: %s, ", "minLength", strconv.FormatInt(int64(t.MinLen), 10))
```

Comment:
> that's confusing. 5.2.2 specifies minLength. Are you sure?

I mean to reply to the following comment, but I might have read it wrong:
> ideally we should limit ourselves to 52bits to be compatible with JavaScript, but not sure what the rules are on that.

The way I read it now, you are talking about limiting the _numeric_ value of minLengt/maxLength to that supported by JavaScript, whic is 52 bits. The JSON or JSON Schema spec does  however not pose such a limit.

```
3.2.  Validation of numeric instances

The JSON specification does not define any bounds to the scale or precision of numeric values. JSON Schema does not define any such bounds either.
```